include/glow/Support/Error.h

```diff
@@ -411,11 +411,10 @@ class GlowError : protected detail::CheckState<detail::enableCheckingErrors> {
     return std::move(errorValue_);
   }
 
-protected:
+public:
```

Comment:
Why does this need to be made public?
It's intentionally protected so that Error has to be initialized through success() or empty()

Reply:
@jackm321 I don't know exactly why there is this difference between Linux build and Windows. But this modification simply fixes the problem which is indicated in the PR comment (somewhere in HostManager.cpp line 180 where std::promise is used). Do you have another solution for this? This is pretty urgent for us right now. Thanks!

Reply:
@jackm321 @jfix71 Any news on this?

Reply:
@mciprian13 does making `std::promise<Error>` a friend class of Error work?
If not in HostManager.cpp:180 you could just change the std::promise<Error> to std::promise<void> and pass an Error in by reference or pointer for example in runNetworkBlocking in HostManager.cpp::671

Reply:
@jackm321 Making ```std::promise<Error>``` friend class of Error does not work (some other classes used inside std::promise seem to use the protected constructor). The second proposal of yours honestly I am not sure I know how do (not familiar with ```std::promise``` and threads at all). I inserted a macro switch to make the constructor public only for the Windows build. If fine by you I would be very grateful to have this landed as is. If not I would appreciate your help in making the second change you mentioned. Thanks!

Reply:
Hi @mciprian13 ok sounds good